### PR TITLE
[EP-535] Added supported card networks

### PR DESCRIPTION
**Pull request by mpetrenco**
- Added ability to set `supportedCardNetworks` to the payment method screen;
- Added new validation logic to handle unsupported networks;
- Added haptic feedback on failed Add Card request;
- Fixed a bug with Secure Code input not working;

### CODE REVIEW

File: Source/JudoKit.h [hunk 1]
```diff
@@ -259,12 +260,14 @@ static NSString *__nonnull const JudoKitVersion = @"8.2.1";
  *  @param amount               The amount and currency of the payment (default is GBP)
  *  @param reference         The reference for this transaction
  *  @param methods              The payment methods to be shown
+ *  @param cardNetworks   The supported card networks to be accepted
```
Comment (gcojocaru) [inline Source/JudoKit.h]: Would be better "The supported card networks" or "Accepted card networks"

File: Source/JudoKit.h [hunk 2]
```diff
@@ -274,12 +277,14 @@ static NSString *__nonnull const JudoKitVersion = @"8.2.1";
  *  @param amount               The amount and currency of the payment (default is GBP)
  *  @param reference            The reference for this transaction
  *  @param methods              The payment methods to be shown
+ *  @param cardNetworks   The supported card networks to be accepted
```
Comment (gcojocaru) [inline Source/JudoKit.h]: Would be better "The supported card networks" or "Accepted card networks"

File: Source/Modules/AddCard/Builder/JPAddCardBuilder.h [hunk 3]
```diff
@@ -34,12 +35,14 @@
  *
  *  @param transaction - a JPTransaction object that describes the transaction details
  *  @param theme - a JPTheme object that primarily defines the style of Add Card view
+ *  @param cardNetworks - the supported card networks to be accepted
```
Comment (gcojocaru) [inline Source/Modules/AddCard/Builder/JPAddCardBuilder.h]: Would be better "The supported card networks" or "Accepted card networks"

File: Source/Modules/AddCard/Interactor/JPAddCardInteractor.h [hunk 4]
```diff
@@ -134,11 +135,13 @@
  *
  *  @param cardValidationService - a JPCardValidationService instance for validating the card details
  *  @param transactionService - a JPTransactionService instance for handling the transaction
+ *  @param cardNetworks - the supported card networks to be accepted
```
Comment (gcojocaru) [inline Source/Modules/AddCard/Interactor/JPAddCardInteractor.h]: Would be better "The supported card networks" or "Accepted card networks"

File: Source/Modules/PaymentMethods/Builder/JPPaymentMethodsBuilder.h [hunk 5]
```diff
@@ -39,13 +40,15 @@
  * @param transitioningDelegate - a transitioning delegate needed for the custom Add Card transition animation
  * @param amount - the amount of the transaction
  * @param reference - the reference for this transaction
+ * @param cardNetworks   The supported card networks to be accepted
```
Comment (gcojocaru) [inline Source/Modules/PaymentMethods/Builder/JPPaymentMethodsBuilder.h]: Would be better "The supported card networks" or "Accepted card networks"

File: Source/Modules/PaymentMethods/Builder/JPPaymentMethodsBuilder.h [hunk 6]
```diff
@@ -56,15 +59,16 @@
...
```
Comment (gcojocaru) [inline Source/Modules/PaymentMethods/Builder/JPPaymentMethodsBuilder.h]: Would be better "The supported card networks" or "Accepted card networks"

File: Source/Modules/PaymentMethods/Router/JPPaymentMethodsRouter.h [hunk 7]
```diff
@@ -59,11 +60,13 @@
  * @param transaction - a reference to the Save Card transaction needed to call the Add Card screen
  * @param transitioningDelegate - the custom transitioning delegate for the Add Card presentation animation
  * @param theme - the JPTheme reference that is used for customizing the Add Card flow
+ * @param cardNetworks - the supported card networks to be accepted
```
Comment (gcojocaru) [inline Source/Modules/PaymentMethods/Router/JPPaymentMethodsRouter.h]: Would be better "The supported card networks" or "Accepted card networks"

File: Source/Extensions/NSError+Judo.h [hunk 8]
```diff
@@ -49,6 +50,7 @@ extern NSString *const JudoErrorDomain;
 + (NSError *)judoMissingChecksumError;
 + (NSError *)judoRequestTimeoutError;
 + (NSError *)judoInvalidCardNumberError;
++ (NSError *)judoUnsupportedCardNetwork:(CardNetwork)cardNetwork;
```
Comment (stpopa) [inline Source/Extensions/NSError+Judo.h]:
```suggestion
+ (NSError *)judoUnsupportedCardNetwork:(CardNetwork)network;
```

File: Source/Extensions/NSError+Judo.m [hunk 9]
```diff
@@ -199,6 +203,18 @@ + (NSError *)judoInvalidCardNumberError {
                            userInfo:@{NSLocalizedDescriptionKey : @"check_card_number".localized}];
 }
 
++ (NSError *)judoUnsupportedCardNetwork:(CardNetwork)cardNetwork {
```
Comment (stpopa) [inline Source/Extensions/NSError+Judo.m]:
```suggestion
+ (NSError *)judoUnsupportedCardNetwork:(CardNetwork)network {
```

**Pull request merged**